Add scrolling support to TextBox

dkrisman opened the pull request:
Adds support for scrolling and appending to text boxes, uses unix2dos for making unix-style newlines seamless to append in text boxes. Adds echo_d example to demo scrolling and appending.

gabdube commented:
Demos makes it so easy to test, thanks! 

Just 1 tiny fix: make `newline-converter` an optional dependency because it's only used by the "textbox" feature.

Also can you check if your changes can be applied to "rich-textbox" too?

dkrisman commented:
I added rich-textbox support and made newline-converter an optional dependency. I added a scroll_lastline function which more accurately targets the last line since rich text boxes behave differently (scrolling the text off of the page) than text boxes when scrolling to a line number beyond the last line. I also modified the len() function to better handle counting CRLFs for getting accurate character indexes for rich text styling. For more advanced programmatic editing of rich text I think the library would need to implement EM_GETTEXTEX and EM_SETTEXTEX messages.

gabdube reviewed (approved):
Good!
